src/main.c

```diff
@@ -184,8 +184,10 @@ static bool parse_args(int argc, char **args)
                                strlen(prog_basename) + 5 + 1);
         assert(prof_out_file);
 
-        sprintf(prof_out_file, "%s/%s%s.prof", cwd_path, rel_path,
-                prog_basename);
+        snprintf(prof_out_file,
+                 strlen(cwd_path) + 1 + strlen(rel_path) +
+                     strlen(prog_basename) + 5 + 1,
+                 "%s/%s%s.prof", cwd_path, rel_path, prog_basename);
```

Comment:
Since we already allocate exactly the required memory for prof_out_file, I don't think there's an actual security issue here, and using snprintf might even be slower. If the goal is just to resolve the macOS build warning, that could be a reasonable justification. However, this approach causes strlen() to be called twice on cwd_path, rel_path, and prog_basename. Maybe we could store the buffer length in an extra variable to avoid relying on the compiler to optimize away the duplicate strlen() calls?